File: src/main/java/club/deneb/client/command/commands/Commands.java
package club.deneb.client.command.commands;

import club.deneb.client.Deneb;
import club.deneb.client.command.Command;
import club.deneb.client.utils.ChatUtil;

/**
 * Created by B_312 on 01/15/21
 */
@Command.Info(command = "commands", description = "Lists all commands.")
public class Commands extends Command {

    @Override
    public void onCall(String s, String[] args) {
        ChatUtil.printChatMessage("\247a" + "Commands:");
        try {
            for (Command cmd : Deneb.getINSTANCE().getCommandManager().commands) {
                if (cmd == this) {
                    continue;
                }
                ChatUtil.printChatMessage("\247a" + cmd.getSyntax().replace("<", "\247e<\2476").replace(">", "\247e>") + "\2477" + " - " + cmd.getDescription());
            }
        } catch (Exception e) {
            ChatUtil.sendNoSpamErrorMessage(getSyntax());
        }
    }

    @Override
    public String getSyntax() {
        return "commands";
    }

}

File: src/main/java/club/deneb/client/command/commands/Commands.kt
package club.deneb.client.command.commands

import club.deneb.client.Deneb
import club.deneb.client.command.Command
import club.deneb.client.utils.ChatUtil.printChatMessage
import club.deneb.client.utils.ChatUtil.sendNoSpamErrorMessage

/**
 * Created by B_312 on 01/15/21
 */
@Command.Info(command = "commands", description = "Lists all commands.")
class Commands : Command() {
    override fun onCall(s: String, vararg args: String) {
        printChatMessage("\u00a7a" + "Commands:")
        try {
            for (cmd in Deneb.getINSTANCE().commandManager.commands) {
                if (cmd === this) {
                    continue
                }
                printChatMessage(
                    "\u00a7a" + cmd.getSyntax().replace("<", "\u00a7e<\u00a76")
                        .replace(">", "\u00a7e>") + "\u00a77" + " - " + cmd.description
                )
            }
        } catch (e: Exception) {
            sendNoSpamErrorMessage(getSyntax())
        }
    }

    override fun getSyntax(): String {
        return "commands"
    }
}